### ISSUE TITLE
Prisms elements and meshes

### ISSUE THREAD

**Issue by gsosajones** [mfem/mfem#807]
Hello,

I downloaded the prisms branch (nc-prism-dev) and everything has been working perfectly in my code, I just have a couple of questions:

1) I am using meshes that are periodic in y. I have a prism mesh that is working, however, the smallest mesh that I have has 18 elements. Since I am doing convergence tests, I have to refine my mesh multiple times which causes my code to be very slow already at the second level of refinement. Is it possible to create a mesh with fewer elements such that it is periodic in y?

2) As far as I understand, the DG basis functions for quads in MFEM is Q^k (maximum degree polynomials), and for triangles, it is P^k (total degree polynomials). How about for prisms?

Thank you for all your help.

Best,
Giselle

**Comment by jakubcerveny** [comment]
Hi Giselle, I believe there need to be at least 3 elements along the periodic dimension (see `data/periodic-square.mesh`). @v-dobrev might want to correct me, but I believe this is to avoid confusion in the edges of the mesh. If you only need periodicity in one dimension, you should be able to define a mesh with 1x3x1 hexes, or in your case 1x3x1x2 prisms. 

As for DG prisms, you can see in `L2_WedgeElement::CalcShape` that the FE is a tensor product of triangle and segment finite elements.

Please note that the branch `nc-prism-dev` is under development and might not be stable.

**Comment by v-dobrev** [comment]
Hi @gsosajones,

With two levels of uniform refinement, you have 18x8x8=1152 elements which is not that many, so there must be something else going on. Are you using very high-order elements? Did you build MFEM with debugging enabled (i.e. with `MFEM_DEBUG=YES`)? Are you running in serial or in parallel?

You can try adding some print statements in your code to see which particular calls become slow as you refine (you can use `mfem::StopWatch` to measure actual timings, if you want).

**Comment by gsosajones** [comment]
Hello,

I am attaching my initial mesh. I could actually take only one row of triangles in the z-direction to make it smaller, and then I would only have 6 elements initially. I tried running example 1 with this mesh and everything seems to be fine, however, I am not getting convergence in my particular application, so maybe there is something wrong in my codes.

I am running everything in serial and I did build MFEM in debug mode. I am going to add some print statements to see if I find where is my code taking too long to run and maybe that will shed some light on to where the mistake is.

Thank you so much.

Best,
Giselle

[periodic3D_prisms.txt](https://github.com/mfem/mfem/files/3008462/periodic3D_prisms.txt)
